src/legacy/server/kbn_server.d.ts

```diff
@@ -101,38 +105,63 @@ export interface PluginsSetup {
   [key: string]: object;
 }
 
+export interface Platform {
+  env: {
+    mode: Readonly<EnvironmentMode>;
+    packageInfo: Readonly<PackageInfo>;
+  };
+  handledConfigPaths: UnwrapPromise<ReturnType<ConfigService['getUsedPaths']>>;
+  setupDeps: {
+    core: CoreSetup;
+    plugins: PluginsSetup;
+  };
+  startDeps: {
+    core: CoreSetup;
+    plugins: Record<string, object>;
+  };
+  __internals: {
+    elasticsearch: LegacyServiceSetupDeps['core']['elasticsearch'];
+    hapiServer: LegacyServiceSetupDeps['core']['http']['server'];
+    kibanaMigrator: LegacyServiceStartDeps['core']['savedObjects']['migrator'];
+    savedObjectsClientProvider: LegacyServiceStartDeps['core']['savedObjects']['clientProvider'];
+    uiPlugins: LegacyServiceSetupDeps['core']['plugins']['uiPlugins'];
+    uiSettings: LegacyServiceSetupDeps['core']['uiSettings'];
+    rendering: LegacyServiceSetupDeps['core']['rendering'];
+  };
+  logger: LoggerFactory;
+}
+
+export interface LegacyPlugins {
+  pluginSpecs: LegacyServiceDiscoverPlugins['pluginSpecs'];
+  disabledPluginSpecs: LegacyServiceDiscoverPlugins['pluginSpecs'];
+  uiExports: LegacyServiceDiscoverPlugins['uiExports'];
+}
+
 // eslint-disable-next-line import/no-default-export
 export default class KbnServer {
   public readonly newPlatform: {
-    __internals: {
-      hapiServer: LegacyServiceSetupDeps['core']['http']['server'];
-      uiPlugins: LegacyServiceSetupDeps['core']['plugins']['uiPlugins'];
-      elasticsearch: LegacyServiceSetupDeps['core']['elasticsearch'];
-      uiSettings: LegacyServiceSetupDeps['core']['uiSettings'];
-      kibanaMigrator: LegacyServiceStartDeps['core']['savedObjects']['migrator'];
-    };
-    env: {
-      mode: Readonly<EnvironmentMode>;
-      packageInfo: Readonly<PackageInfo>;
-    };
+    __internals: Omit<Platform['__internals'], 'savedObjectsClientProvider'>;
```

Comment:
Why we have to do this? Cannot we just consume `Platform['__internals']` ? Are there any other `Platform` type consumers?